validator/src/main/java/com/google/daq/mqtt/sequencer/sequences/ConfigSequences.java

```diff
@@ -22,6 +27,39 @@
  */
 public class ConfigSequences extends SequenceRunner {
 
+  @Test
+  @Description("Push endpoint config message to device that results in a connection error.")
+  public void endpoint_config_connection_error() {
+    // blob _iot_endpoint_config = ...
+    BlobBlobsetConfig cfg = new BlobBlobsetConfig();
+    cfg.phase = BlobPhase.FINAL;
+    // { protocol=mqtt, client_id=test_project/device; hostname=localhost }
+    cfg.base64 = "ewogICJwcm90b2NvbCI6ICJtcXR0IiwKICAiY2xpZW50X2lkIjogInRlc3RfcHJvamVjdC9kZXZp"
+        + "Y2UiLAogICJob3N0bmFtZSI6ICJsb2NhbGhvc3QiCn0K";
+    cfg.content_type = "application/json";
+    deviceConfig.blobset = new BlobsetConfig();
+    deviceConfig.blobset.blobs = new HashMap<String, BlobBlobsetConfig>();
+    deviceConfig.blobset.blobs.put("_iot_endpoint_config", cfg);
+
+    updateConfig();
+    untilTrue("pubber has attempted endpoint reconfig", () -> {
```

Comment:
it's not always pubber -- this might be running against a manufacturer device

Reply:
Right PTAL